asynciterator.ts

```diff
@@ -309,6 +309,39 @@ export class AsyncIterator<T> extends EventEmitter {
     return '';
   }
 
+  /**
+    Consume all remaining items of the iterator into an array that will be returned asynchronously.
+    @param {object} [options] Settings for array creation
+    @param {integer} [options.limit] The maximum number of items to place in the array.
+   */
+  toArray(options?: { limit?: number }): Promise<T[]> {
+    return new Promise<T[]>((resolve, reject) => {
+      // Collect items in an array
+      const items: T[] = [];
+
+      // Early return if the limit is invalid
+      // @ts-ignore
+      if (options?.limit === 0)
```

Comment:
Or negative (specifically `-0`). `limit` might be the result of a calculation so devs might not always see.